ext/node/polyfills/http.ts

```diff
@@ -721,8 +670,20 @@ class ServerImpl extends EventEmitter {
             break;
           }
           const req = new IncomingMessageForServer(reqEvent.request);
-          const res = new ServerResponse(reqEvent, undefined);
-          this.emit("request", req, res);
+          if (req.upgrade && this.listenerCount("upgrade") > 0) {
+            const conn = await denoHttp.upgradeHttpRaw(
+              reqEvent.request,
+              tcpConn,
+            ) as Deno.Conn;
+            const socket = new Socket({
+              handle: new TCP(constants.SERVER, conn),
+            });
+            this.emit("upgrade", req, socket, Buffer.from([]));
+            return;
+          } else {
+            const res = new ServerResponse(reqEvent);
+            this.emit("request", req, res);
+          }
```

Comment:
Looks great. Is there an immutable node buffer we can use to avoid allocating? It's only on connection so probably isn't a huge deal.

Reply:
I'm not sure about that. Are you talking about the `Buffer.from([])`?